src/plugins/github/fetchGithubRepo.js

```diff
@@ -34,7 +46,7 @@ import {validateToken} from "./token";
  */
 export default async function fetchGithubRepo(
   repoId: RepoId,
-  options: {|+token: string, +cacheDirectory: string|}
+  options: {|+token: string, +cacheDirectory: string, +cache?: CacheProvider|}
```

Comment:
This signature where we provide a (required) cacheDirectory, an (optional) cacheProvider, and then ignore the required argument if the optional one is provided, is really weird.

If you want to make a minimal change to support both use cases, you could set up a union type, right?

```js
type CacheInfo = {|type: "DIRECTORY", directory: string|} 
  | {|type: "PROVIDER", provider: CacheProvider|}
```